### PR TITLE
Added support for running FBS on top of Pegasus

### PR DESCRIPTION
**Pull request by anton164**
- Infer tokenization max length from tokenizer (pegasus is pretrained with 512, bart with 1024)
- Refactor Phrase logits processor & backtracking to work on a word-level since Pegasus tokenizer doesn't include spaces per token
- Added tests for Pegasus banned phrases
- Added debug results for Pegasus
- Fixed bug in handling of padded beams

### CODE REVIEW

File: results/debug-oracle.json [hunk 1]
```diff
@@ -153,7 +153,7 @@
           "generation_metadata": {
             "score": -0.48298680782318115,
             "dropped_seqs": [],
-            "n_words_checked": 77
+            "n_words_checked": 65
```
Comment (anton164) [inline results/debug-oracle.json:156]: This has been reduced due to the bugfix in this commit: https://github.com/danton-nlp/factual-beam-search/pull/50/commits/f238b1b66f46312738c69c4bbe70d098ff0621ec

File: src/word_logits_processor.py [hunk 2]
```diff
@@ -36,74 +32,78 @@ def __init__(self, tokenizer, num_beams, word_validator: WordValidator):
...
-
-        # Word-level checks
-        current_subword = self.tokenizer.decode(token_id)
-        backtrack_word = ""
-        is_subword_ending = False
-        for char in current_subword:
+
+        # Begin by checking whether tokens
+        # to-be-generated indicate a phrase ending
+        if "pegasus" in self.tokenizer.name_or_path:
```
Comment (anton164) [inline src/word_logits_processor.py:56]: Ended up doing special handling for pegasus - otherwise we do unnecessary checks for BART.

**Review by dleve123 — APPROVED** [review]
LGTM!
